pipenv/project.py

```diff
@@ -685,8 +685,8 @@ def create_pipfile(self, python=None):
         from .vendor.pip_shims.shims import (
             ConfigOptionParser, make_option_group, index_group
         )
-
-        config_parser = ConfigOptionParser(name=self.name)
+        # Inherit the pip's index configuration of install command.
+        config_parser = ConfigOptionParser(name="install")
```

Comment:
Is this change intended, @frostming?  It doesn't seem related to the previous root-path handling modification (#3386)

Reply:
(to clarify, on taking a second look - this change does appear to affect a similar section of code to the previous modifications, but it's not immediately clear whether it's a direct revert of them; it may be the correct way to go, but I figured it'd be worth checking just in case)

Reply:
The original issue is that when under the root directory, `ConfigOptionParser` will complain that `sel.name` is empty. The fix in #3386 prevented it fron happening at early stage, which brought a breakage in many users' docker building setup.

So I revert the early checking and instead make `ConfigOptionaParser` happy by passing a meaningful section name to it.
Because giving an arbitrary name doesn't make any sense, which i didn't realize before. This should be the correct path to resolving the issue.

Reply:
Thank you for the explanation!